src/box/lua/report_daemon.lua

```diff
@@ -0,0 +1,86 @@
+-- report_daemon.lua (internal file)
+
+local log   = require('log')
+local json  = require('json')
+local fiber = require('fiber')
+local http  = require('http.client')
+
+local PREFIX = "report_daemon"
+
+local daemon = {
+    reports_enabled  = false,
+    reports_interval = 0,
+    reports_host     = nil,
+    fiber            = nil,
+    control          = nil
+}
+
+local function fill_in_report(report)
+    report.tarantool_version = box.info.version
+    report.server_id         = box.info.uuid
+    report.cluster_id        = box.info.cluster.uuid
+    report.timestamp         = fiber.time64()
+    return report
+end
+
+local function version_report(self)
```

Comment:
Every line in this function can throw. How about a plan for it? You could unit-test it btw, using tap, by exporting part of the api of the daemon via box.internal, and adding a test to box-tap/ suite